src/components/core/dropdown/dropdown-base/list-items.tsx

```diff
@@ -0,0 +1,153 @@
+    },
+
+    compoundVariants: [
+        {
+            dark: true,
+            active: true,
+            css: {
+                backgroundColor: '$greyDark600',
+            },
+        },
+        {
+            dark: true,
+            selected: true,
+            css: {
+                backgroundColor: '$greyDark500',
+            },
+        },
+        {
+            is_text_string_type: false,
+            selected: true,
+            css: {
+                fontWeight: '$regular',
+            },
+        },
+    ],
+});
+
+const Item = ({
+    child_ref,
+    className,
+    dark,
+    is_active,
+    is_align_text_center,
+    item,
+    onItemSelection,
+    selected_value,
+}: TItem) => {
+    return (
+        <ItemContainer
+            active={is_active}
+            className={classNames(className, {
+                nohover: selected_value === item.value || item.disabled,
+            })}
+            dark={dark}
+            data-testid="dt_list_item"
+            disbaled={item.disabled}
+            is_text_string_type={typeof item.text === 'string'}
+            onClick={() => onItemSelection(item)}
+            ref={child_ref}
+            selected={selected_value === item.value}
+        >
+            <Text
+                align={is_align_text_center ? 'center' : 'left'}
+                color={dark && selected_value === item.value ? 'prominent' : 'general'}
+                css={{ margin: 0 }}
+                type="paragraph-2"
+            >
+                {item.text}
+            </Text>
+        </ItemContainer>
+    );
+};
+
+const ListItems = React.forwardRef<React.ElementRef<typeof ItemContainer>, TListItems>(
+    (
+        { active_index, className, dark, is_align_text_center, list, not_found_text, onItemSelection, selected_value },
+        ref,
+    ) => {
+        return (
+            <>
+                {list.length ? (
+                    list.map((item, item_idx) => (
+                        <Item
+                            child_ref={item_idx === active_index ? ref : null}
+                            className={className}
+                            dark={dark}
+                            is_align_text_center={is_align_text_center}
+                            key={item_idx}
+                            item={item}
+                            is_active={item_idx === active_index}
+                            onItemSelection={onItemSelection}
+                            selected_value={selected_value}
+                        />
+                    ))
+                ) : (
+                    <Item
+                        className={className}
+                        dark={dark}
+                        is_align_text_center={is_align_text_center}
+                        item={{ value: '', text: not_found_text }}
+                        onItemSelection={() => onItemSelection({ value: '', text: '' })}
+                    />
+                )}
+            </>
+        );
```

Comment:
Early return is more readable than nested logic like this

```suggestion
   if (list.length)
       return <></>
       
  return <Item />
  ```